# Actividad Evaluable: Obtención de estadisticas descriptivas

Elaborado por: Jorge Sánchez Ponce A07106589

In [ ]:
# Importamos las librerías

import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

# Cargamos el archivo y lo convertimos en un data frame
sputy = pd.read_csv("C:\\Users\\georg\\REPO_SEMANA_TEC\\data\\spotify.csv")
sputy.head()


Unnamed: 0                track_id                 artists  \
0           0  5SuOikwiRyPMVoIQDJUgSV             Gen Hoshino   
1           1  4qPNDBW1i3p13qLCt0Ki3A            Ben Woodward   
2           2  1iJBSr7s7jYXzM8EGcbK5b  Ingrid Michaelson;ZAYN   
3           3  6lfxq3CG4xtTiEg7opyCyx            Kina Grannis   
4           4  5vjLSffimiIP26QG5WcN2K        Chord Overstreet   

                                          album_name  \
0                                             Comedy   
1                                   Ghost (Acoustic)   
2                                     To Begin Again   
3  Crazy Rich Asians (Original Motion Picture Sou...   
4                                            Hold On   

                   track_name  popularity  duration_ms  explicit  \
0                      Comedy          73       230666     False   
1            Ghost - Acoustic          55       149610     False   
2              To Begin Again          57       210826     False   
3  Can't Help Falling In Love          71       201933     False   
4                     Hold On          82       198853     False   

   danceability  energy  ...  loudness  mode  speechiness  acousticness  \
0         0.676  0.4610  ...    -6.746     0       0.1430        0.0322   
1         0.420  0.1660  ...   -17.235     1       0.0763        0.9240   
2         0.438  0.3590  ...    -9.734     1       0.0557        0.2100   
3         0.266  0.0596  ...   -18.515     1       0.0363        0.9050   
4         0.618  0.4430  ...    -9.681     1       0.0526        0.4690   

   instrumentalness  liveness  valence    tempo  time_signature  track_genre  
0          0.000001    0.3580    0.715   87.917               4     acoustic  
1          0.000006    0.1010    0.267   77.489               4     acoustic  
2          0.000000    0.1170    0.120   76.332               4     acoustic  
3          0.000071    0.1320    0.143  181.740               3     acoustic  
4          0.000000    0.0829    0.167  119.949               4     acoustic  

[5 rows x 21 columns]

Utilizamos el atributo ```shape``` para saber la dimensión que tiene nuestro data frame, es decir, la cantidad de filas y columnas.

In [13]:
sputy.shape

(114000, 21)

Como se observa, el df (data frame pero para fines más prácticos escribiremos únicamente df) tiene $114000$ filas y $21$ columnas, esto último indica que contamos con $21$ variables.

Utilizamos el método ```info``` para obtener el nombre, la cantidad de valores y el tipo de datos de cada variable (columna) que posee nuestro df.

In [6]:
sputy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114000 entries, 0 to 113999
Data columns (total 21 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Unnamed: 0        114000 non-null  int64  
 1   track_id          114000 non-null  object 
 2   artists           113999 non-null  object 
 3   album_name        113999 non-null  object 
 4   track_name        113999 non-null  object 
 5   popularity        114000 non-null  int64  
 6   duration_ms       114000 non-null  int64  
 7   explicit          114000 non-null  bool   
 8   danceability      114000 non-null  float64
 9   energy            114000 non-null  float64
 10  key               114000 non-null  int64  
 11  loudness          114000 non-null  float64
 12  mode              114000 non-null  int64  
 13  speechiness       114000 non-null  float64
 14  acousticness      114000 non-null  float64
 15  instrumentalness  114000 non-null  float64
 16  liveness          11

Aquí se observa que la gran mayoría de nuestras variables son númericas, a excepción de las variables ```track_id```, ```album_name```, ```track_name``` y ```track_genre``` que son del tipo ```object```, y la variable ```explicit```, que es del tipo ```bool```, esto quiere decir, que son variables categóricas, y más específicamente, son variables nominales, dado que no existe ningún tipo de jerarquía u orden en ellas.

In [10]:
sputy.describe()

Unnamed: 0     popularity   duration_ms   danceability  \
count  114000.000000  114000.000000  1.140000e+05  114000.000000   
mean    56999.500000      33.238535  2.280292e+05       0.566800   
std     32909.109681      22.305078  1.072977e+05       0.173542   
min         0.000000       0.000000  0.000000e+00       0.000000   
25%     28499.750000      17.000000  1.740660e+05       0.456000   
50%     56999.500000      35.000000  2.129060e+05       0.580000   
75%     85499.250000      50.000000  2.615060e+05       0.695000   
max    113999.000000     100.000000  5.237295e+06       0.985000   

              energy            key       loudness           mode  \
count  114000.000000  114000.000000  114000.000000  114000.000000   
mean        0.641383       5.309140      -8.258960       0.637553   
std         0.251529       3.559987       5.029337       0.480709   
min         0.000000       0.000000     -49.531000       0.000000   
25%         0.472000       2.000000     -10.013000       0.000000   
50%         0.685000       5.000000      -7.004000       1.000000   
75%         0.854000       8.000000      -5.003000       1.000000   
max         1.000000      11.000000       4.532000       1.000000   

         speechiness   acousticness  instrumentalness       liveness  \
count  114000.000000  114000.000000     114000.000000  114000.000000   
mean        0.084652       0.314910          0.156050       0.213553   
std         0.105732       0.332523          0.309555       0.190378   
min         0.000000       0.000000          0.000000       0.000000   
25%         0.035900       0.016900          0.000000       0.098000   
50%         0.048900       0.169000          0.000042       0.132000   
75%         0.084500       0.598000          0.049000       0.273000   
max         0.965000       0.996000          1.000000       1.000000   

             valence          tempo  time_signature  
count  114000.000000  114000.000000   114000.000000  
mean        0.474068     122.147837        3.904035  
std         0.259261      29.978197        0.432621  
min         0.000000       0.000000        0.000000  
25%         0.260000      99.218750        4.000000  
50%         0.464000     122.017000        4.000000  
75%         0.683000     140.071000        4.000000  
max         0.995000     243.372000        5.000000

Ahora utilizamos el método ```describe``` para mostrar las estadísticas requeridas, cabe señalar, que este método solo evalúa las variables numéricas, las variables categóricas se evaluarán próximamente.

- La mayoría de las variables tiene un mínimo de 0 a excepción de la variable ```loudness```, cuyo mínimo es -49.531.
- Muchas variables parecen tener un rango de 0-1, lo que puede indicar un porcentaje.
- La popularidad tiene un rango de 0-100 con una media de 33.238, de lo que se infiere que la mayoría de las canciones no son tan populares al parecer.
- La duración promedio de una canción es de aproximadamente de 3.8 minutos, lo que equivale a 3 minutos y 48 segundos, un tiempo esperado y congruente con las canciones actuales.
- El promedio del tempo es de 122.147 pulsaciones por minuto, aproximadamente 2 pulsos por segundo.
- La mediana (percentil 50) en casi todas las variables se asemeja bastante al promedio, lo que indica que existe una distribución uniforme o muy cercana a esta.

In [11]:
cat_var = sputy.describe(include=['object'])
cat_var

track_id      artists                  album_name  \
count                   114000       113999                      113999   
unique                   89741        31437                       46589   
top     6S3JlDAGk3uu3NtZbPnuhS  The Beatles  Alternative Christmas 2022   
freq                         9          279                         195   

             track_name track_genre  
count            113999      114000  
unique            73608         114  
top     Run Rudolph Run    acoustic  
freq                151        1000

En esta tabla se muestran algunas características de las variables categóricas, exceptuando la variable ```explicit``` puesto que es del tipo ```bool``` y solo contiene valores de ```True``` y ```False```, por lo que para este ejercicio no es de nuestro interés.

Se muestra que el artista con más observaciones es The Beatles, con 279 registros, el albúm con más observaciones es Alternative Christmas 2022 con 195, la canción con más observaciones es Run Rudolph Run con 151, y por último, el género más popular parecer ser el acústico con 1000 registros, pero evaluaremos estas variables más a fondo.

También se expone la cantidad total de artistas, álbumes, canciones y géneros diferentes, siendo 31437 artistas, 46589 álbumes, 73608 canciones y 114 géneros.

In [26]:
artistas = sputy['artists'].nunique()
album = sputy['album_name'].nunique()
track = sputy['track_name'].nunique()
genre = sputy['track_genre'].nunique()

In [27]:
print(f"El número total de artistas (grupos y singulares) son: {artistas}")
print(f"El número total de álbumes son: {album}")
print(f"El número total de canciones son: {track}")
print(f"El número total de géneros son: {genre}")

El número total de artistas (grupos y singulares) son: 31437
El número total de álbumes son: 46589
El número total de canciones son: 73608
El número total de géneros son: 114


Esto simplemente nos confirma lo que anteriormente se había mencionado, ahora queremos analizar el top 10 de artistas, álbumes, canciones y géneros más populares.

In [28]:
print("\nArtistas más frecuentes (Top 10)")
print(sputy['artists'].value_counts().head(10))


Artistas más frecuentes (Top 10)
artists
The Beatles        279
George Jones       271
Stevie Wonder      236
Linkin Park        224
Ella Fitzgerald    222
Prateek Kuhad      217
Feid               202
Chuck Berry        190
Håkan Hellström    183
OneRepublic        181
Name: count, dtype: int64


En primer lugar aparece The Beatles, seguido de manera muy cercana por George Jones y después Stevie Wonder.

In [29]:
print("\nÁlbumes más frecuentes (Top 10)")
print(sputy['album_name'].value_counts().head(10))


Álbumes más frecuentes (Top 10)
album_name
Alternative Christmas 2022     195
Feliz Cumpleaños con Perreo    184
Metal                          143
Halloween con perreito         123
Halloween Party 2022           115
The Complete Hank Williams     111
Fiesta portatil                110
Frescura y Perreo              106
Esto me suena a Farra          105
Perreo en Halloween            103
Name: count, dtype: int64


El álbum más popular es Alternative Christmas 2022, seguido de Feliz Cumpleaños con Perreo y en tercer lugar Metal.

In [30]:
print("\nCanciones más frecuentes (Top 10)")
print(sputy['track_name'].value_counts().head(10))


Canciones más frecuentes (Top 10)
track_name
Run Rudolph Run                   151
Halloween                          88
Frosty The Snowman                 81
Little Saint Nick - 1991 Remix     76
Last Last                          75
Christmas Time                     72
CÓMO SE SIENTE - Remix             64
Sleigh Ride                        61
RUMBATÓN                           60
X ÚLTIMA VEZ                       58
Name: count, dtype: int64


Aquí quisiera mencionar que de las 10 canciones más populares, 4 de ellas se relacionan con el tema de la Navidad, estas canciones son: Run Rudolph Run (1), Frosty The Snowman (3), Christmas Time (6) y Sleigh Ride (8).

El número entre paréntesis indica el lugar en la lista.

In [31]:
print("\nGéneros más frecuentes (Top 10)")
print(sputy['track_genre'].value_counts().head(10))


Géneros más frecuentes (Top 10)
track_genre
acoustic       1000
afrobeat       1000
alt-rock       1000
alternative    1000
ambient        1000
anime          1000
black-metal    1000
bluegrass      1000
blues          1000
brazil         1000
Name: count, dtype: int64


Al analizar esta sección ocurre algo interesante, dado que los 10 géneros con más observaciones tienen la misma cantidad de registros, que son 1000, y ya que previamente se mencionó que el número total de géneros eran 114, podemos inferir que todos los géneros que se presentan tendrán un acumulado total de 1000 registros cada uno, generando un total final de 114,000.

In [34]:
print(sputy['track_genre'].value_counts().head(114))

track_genre
acoustic       1000
afrobeat       1000
alt-rock       1000
alternative    1000
ambient        1000
               ... 
techno         1000
trance         1000
trip-hop       1000
turkish        1000
world-music    1000
Name: count, Length: 114, dtype: int64


Aquí se confirma la hipótesis formulada, por lo cual realmente no podemos realizar algún análisis significativo sobre esta variable.